src/mod/asr_tts/mod_unimrcp/mod_unimrcp.c

```diff
@@ -1962,9 +1966,9 @@ static apt_bool_t speech_on_channel_add(mrcp_application_t *application, mrcp_se
 
 	/* notify of channel open */
 	if (globals.enable_profile_events && switch_event_create_subclass(&event, SWITCH_EVENT_CUSTOM, MY_EVENT_PROFILE_OPEN) == SWITCH_STATUS_SUCCESS) {
-		orig_session = switch_core_session_locate(schannel->session_uuid);
-		if (orig_session && switch_core_session_read_lock(orig_session) == SWITCH_STATUS_SUCCESS) {
+		if ((orig_session = switch_core_session_locate(schannel->session_uuid))) {
 			orig_channel = switch_core_session_get_channel(orig_session);
+			switch_core_session_rwunlock(orig_session);
```

Comment:
We need to call unlock at the bottom, channel can be destroyed outside of the session lock and may cause crash. Lock the session when you are still using the channel.

Reply:
Updated